### PR TITLE
STM32WB IWDG Support

### PR DESCRIPTION
**Pull request by robbawebba**
The STM32WB SoC supports an independent watchdog that is compatible with the existing stm32 watchdog driver and watchdog API. This pull request adds watchdog support for this SoC for use with the watchdog drivers as well as enables watchdog support for the nucleo_wb55rg board.

This has been tested with the nucleo_wb55rg and the sample located at `samples/drivers/watchdog`.

### CODE REVIEW

**Comment by robbawebba** [comment]
I noticed that the file `boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst` does not contain an underscore in the board name (i.e. `nucleo_wb55rg.rst`). It doesn't seem like too big of an issue, but I wanted to double check with you in case either of you had a preference. Is this alright, or is this something that you would like changed?

File: soc/arm/st_stm32/stm32wb/dts_fixup.h [hunk 1]
```diff
@@ -155,4 +155,6 @@
 #define DT_ADC_1_CLOCK_BITS		DT_ST_STM32_ADC_50040000_CLOCK_BITS_0
 #define DT_ADC_1_CLOCK_BUS		DT_ST_STM32_ADC_50040000_CLOCK_BUS_0
 
+#define DT_WDT_0_NAME         DT_INST_0_ST_STM32_WATCHDOG_LABEL
```
Comment (erwango) [inline soc/arm/st_stm32/stm32wb/dts_fixup.h]: Please keep indentation

File: soc/arm/st_stm32/stm32wb/dts_fixup.h [hunk 2]
```diff
@@ -155,6 +155,6 @@
 #define DT_ADC_1_CLOCK_BITS		DT_ST_STM32_ADC_50040000_CLOCK_BITS_0
 #define DT_ADC_1_CLOCK_BUS		DT_ST_STM32_ADC_50040000_CLOCK_BUS_0
 
-#define DT_WDT_0_NAME         DT_INST_0_ST_STM32_WATCHDOG_LABEL
+#define DT_WDT_0_NAME			DT_INST_0_ST_STM32_WATCHDOG_LABEL
```
Comment (erwango) [inline soc/arm/st_stm32/stm32wb/dts_fixup.h]: Well, move this fix in previous commit and we're good

**Review by erwango — CHANGES_REQUESTED** [review]
Minor remark

**Review by dbkinder — CHANGES_REQUESTED** [review]
It would indeed be good to rename the document to ``nucleo_wb55rg.rst``, otherwise, LGTM.

File: soc/arm/st_stm32/stm32wb/dts_fixup.h (hunk 2 above)
Comment (robbawebba) [reply soc/arm/st_stm32/stm32wb/dts_fixup.h]: Addressed, thanks for catching that!

File: soc/arm/st_stm32/stm32wb/dts_fixup.h (hunk 1 above)
Comment (robbawebba) [reply soc/arm/st_stm32/stm32wb/dts_fixup.h]: Addressed, thanks!

File: soc/arm/st_stm32/stm32wb/dts_fixup.h (hunk 2 above)
Comment (robbawebba) [reply soc/arm/st_stm32/stm32wb/dts_fixup.h]: Addressed, thanks!

**Comment by robbawebba** [comment]
@dbkinder and @erwango Thanks for the feedback! I made the requested changes. Could you please check that the changes meet your expectations? Thanks in advance!

**Review by erwango — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks.

**Pull request merged**